File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<int>stk;
        for(int i=0;i<s.size();i++){
            
                if(stk.size()==0 and (s[i]==')' or s[i]==']' or s[i]=='}')){
                        return false;
                }
                else if(s[i]=='(' or s[i]=='[' or s[i]=='{'){
                        stk.push(s[i]);
                }
                
                else if(s[i]==')' and stk.top()=='('){
                        stk.pop();    
                }
            
                else if(s[i]==']' and stk.top()=='['){
                        stk.pop();    
                }
            
                else if(s[i]=='}' and stk.top()=='{'){
                        stk.pop();    
                }
            
                else{
                            // cout<<s[i]<<endl;
                            return false;
                    
                }    
            
        }
        
        return stk.size()==0;
        
        
        
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        stack=[]
        
        for item in s:
            
            if item=="(" or item=="[" or item=="{":
                
                stack.append(item)
                
                
            else:
                
                if not len(stack):
                        return False
                    
                if item==')' and stack[-1]=='(':
                            stack.pop();
                            
                
                elif item==']' and stack[-1]=='[':
                        
                            stack.pop();
                            
            
                elif item=='}' and stack[-1]=='{':
                        
                            stack.pop();
                
                else:
                        return False
                    
            
        return len(stack)==0      
            
            
        